cmd/basecli/commands/cmds.go

```diff
@@ -103,39 +111,53 @@ func readSendTxFlags(tx *btypes.SendTx) error {
 
 /******** AppTx *********/
 
+// BroadcastAppTx wraps, signs, and executes an app tx basecoin transaction
+func BroadcastAppTx(tx *btypes.AppTx) (*ctypes.ResultBroadcastTxCommit, error) {
+
+	// Generate app transaction to be broadcast
+	appTx := WrapAppTx(tx)
+	appTx.AddSigner(txcmd.GetSigner())
+
+	// Sign if needed and post to the node.  This it the work-horse
+	return txcmd.SignAndPostTx(appTx)
+}
+
+// AddAppTxFlags adds flags required by apptx
 func AddAppTxFlags(fs *flag.FlagSet) {
-	fs.String(AmountFlag, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
-	fs.String(FeeFlag, "0mycoin", "Coins for the transaction fee of the format <amt><coin>")
-	fs.Int64(GasFlag, 0, "Amount of gas for this transaction")
-	fs.Int(SequenceFlag, -1, "Sequence number for this transaction")
+	fs.String(FlagAmount, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
+	fs.String(FlagFee, "0mycoin", "Coins for the transaction fee of the format <amt><coin>")
+	fs.Int64(FlagGas, 0, "Amount of gas for this transaction")
+	fs.Int(FlagSequence, -1, "Sequence number for this transaction")
 }
 
 // ReadAppTxFlags reads in the standard flags
 // your command should parse info to set tx.Name and tx.Data
-func ReadAppTxFlags(tx *btypes.AppTx) error {
-	//parse the fee and amounts into coin types
-	var err error
-	tx.Fee, err = btypes.ParseCoin(viper.GetString(FeeFlag))
+func ReadAppTxFlags() (gas int64, fee btypes.Coin, txInput btypes.TxInput, err error) {
```

Comment:
great argument.  glad you are putting this to the test to make a better api. more useful and less magic